models/helpers.ts

```diff
@@ -248,6 +248,45 @@ export function getInvoiceStatus(doc: RenderData | Doc): InvoiceStatus {
   return 'Saved';
 }
 
+export function getSerialNoStatusColumn(): ColumnConfig {
+  return {
+    label: t`Status`,
+    fieldname: 'status',
+    fieldtype: 'Select',
+    render(doc) {
+      const status = doc.status as SerialNoStatus;
```

Comment:
Don't use type assertions. Narrow the type by either using predicate functions or conditionals. [ref](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

If `doc.status` is not a `string`, let alone a `SerialNoStatus`, downstream code will have invalid states, color will be undefined.